examples/mnist/mnist.py

```diff
@@ -61,27 +62,39 @@ def run(train_batch_size, val_batch_size, epochs, lr, momentum, log_interval):
     @trainer.on(Events.ITERATION_COMPLETED)
     def log_training_loss(engine):
         iter = (engine.state.iteration - 1) % len(train_loader) + 1
-        if iter % log_interval == 0:
-            print("Epoch[{}] Iteration[{}/{}] Loss: {:.2f}"
-                  "".format(engine.state.epoch, iter, len(train_loader), engine.state.output))
+        desc = "ITERATION - loss: {:.2f}".format(engine.state.output)
+
+        if iter == 1:
```

Comment:
@rave78 could we avoid if check like this ?

Reply:
Another option is to use `if not hasattr(engine.state, 'pdbar_iter')` instead of `if iter == 1`.  Note that `tqdm` is required for the first iteration for each epoch and `tqdm.close()` for the last iteration.

Reply:
is it necessary to add `pdbar_iter` to `engine.state` ? Maybe you can define a `pbar = tqdm(...)` outside the handler like it is done for the [writer in tensorboard example](https://github.com/pytorch/ignite/blob/master/examples/mnist/mnist_with_tensorboardx.py#L80).

Reply:
I see.  Yes, it is possible reducing more the interaction between `tqdm` and the handler.